src/main/java/io/github/railroad/menu/TopMenu.java

```diff
@@ -0,0 +1,238 @@
+package io.github.railroad.menu;
+
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuBar;
+import javafx.scene.control.SeparatorMenuItem;
+
+import static io.github.railroad.lang.LangManager.ENGLISH;
+import static io.github.railroad.lang.LangManager.getLocalization;
+import static io.github.railroad.utility.Components.MenuItemFactory.makeMenuItem;
+import static io.github.railroad.utility.Components.createImage;
+import static io.github.railroad.utility.FileChoosers.createNewGenericFile;
+import static io.github.railroad.utility.FileChoosers.createNewJavaFile;
+import static io.github.railroad.utility.Templates.JavaTemplate.*;
+import static io.github.railroad.utility.Terminal.openTerminal;
+
+/**
+ * Creates all of the items of the top menu.
+ *
+ * @author TheOnlyTails
+ */
+public final class TopMenu extends MenuBar {
+    /**
+     * Creates the top menu and adds all of the submenus.
+     *
+     * @author TheOnlyTails
+     */
+    public void createMenu() {
+        final var fileMenu = new Menu(getLocalization("menu.file", ENGLISH));
+        createFileMenu(fileMenu);
+
+        final var editMenu = new Menu(getLocalization("menu.edit", ENGLISH));
+        createEditMenu(editMenu);
+
+        final var searchMenu = new Menu(getLocalization("menu.search", ENGLISH));
+        createSearchMenu(searchMenu);
+
+        final var runMenu = new Menu(getLocalization("menu.run", ENGLISH));
```

Comment:
Does the Run menu not need a class like the others?
Should it be beneath the others in that case, to provide clear separation?

Reply:
It doesn't contain anything yet, so no.